src/app/shared/components/filters-list/searchbar/searchbar.component.ts

```diff
@@ -1,33 +1,54 @@
-import { Component, OnInit } from '@angular/core'
+import { Component, OnDestroy, OnInit } from '@angular/core'
 import { FormControl, Validators } from '@angular/forms';
-import { Store } from '@ngxs/store';
-import { Subject } from 'rxjs';
+import { Router } from '@angular/router';
+import { Select, Store } from '@ngxs/store';
+import { Observable, Subject } from 'rxjs';
 import { debounceTime, distinctUntilChanged, startWith, takeUntil } from 'rxjs/operators';
+import { NavBarName } from 'src/app/shared/enum/navigation-bar';
+import { Navigation } from 'src/app/shared/models/navigation.model';
 import { SetSearchQueryValue } from 'src/app/shared/store/filter.actions';
+import { NavigationState } from 'src/app/shared/store/navigation.state';
 @Component({
   selector: 'app-searchbar',
   templateUrl: './searchbar.component.html',
   styleUrls: ['./searchbar.component.scss']
 })
-export class SearchbarComponent implements OnInit {
+export class SearchbarComponent implements OnInit, OnDestroy {
+
+  constructor(
+    private store: Store,
+    private router: Router,
+  ) { }
 
-  constructor(private store: Store) { }
   searchValue = new FormControl('', [Validators.maxLength(200), this.searchValidator]);
-  destroy$: Subject<boolean> = new Subject<boolean>();
+  isResultPage: boolean = false;
 
+  @Select(NavigationState.navigationPaths)
+  navigationPaths$: Observable<Navigation[]>;
+  destroy$: Subject<boolean> = new Subject<boolean>();
 
   ngOnInit(): void {
+    this.navigationPaths$.pipe(
+      takeUntil(this.destroy$),
+    ).subscribe((navigationPaths: Navigation[]) => this.isResultPage = navigationPaths.some((path: Navigation) => path.name === NavBarName.TopWorkshops));
+
     this.searchValue.valueChanges
       .pipe(
         takeUntil(this.destroy$),
         debounceTime(300),
         distinctUntilChanged(),
         startWith('')
       ).subscribe(val => {
-        this.store.dispatch(new SetSearchQueryValue(val || ''));
+        !this.isResultPage && this.store.dispatch(new SetSearchQueryValue(val || ''));
```

Comment:
JFYI: !val && func() === val || func()

Reply:
honestly, I didn't get it:)
!val && func() === val || func()

val you mean this.isResultPage or val as from FormControl?